### PR TITLE
22943 update models to use new db_versioning

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#22943
*Issue #:* /bcgov/entity#22942

*Description of changes:*

Update the models and VersionedBusinessDetailsService to use the new DB versioning.

- ***Get the Versioning DB***
Old: `amalgamating_businesses_version = version_class(AmalgamatingBusiness)`
New: `amalgamating_businesses_version = VersioningProxy.version_class(db.session(), AmalgamatingBusiness)`

- ***Get the transaction in the Test***
Old: `uow = versioning_manager.unit_of_work(session).  transaction = uow.create_transaction(session)`
New: `transaction_id = VersioningProxy.get_transaction_id(session())`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/models/amalgamating_business.py [hunk 1]
```diff
@@ -62,7 +61,7 @@ def save(self):
     def get_revision(cls, transaction_id, amalgamation_id):
         """Get amalgamating businesses for the given transaction id."""
         # pylint: disable=singleton-comparison;
-        amalgamating_businesses_version = version_class(AmalgamatingBusiness)
+        amalgamating_businesses_version = VersioningProxy.version_class(db.session(), AmalgamatingBusiness)
         amalgamating_businesses = db.session.query(amalgamating_businesses_version) \
             .filter(amalgamating_businesses_version.transaction_id <= transaction_id) \
             .filter(amalgamating_businesses_version.operation_type == 0) \
```
[the remark this answers is not in the document]
Comment (kzdev420) [reply legal-api/src/legal_api/models/amalgamating_business.py:67]: I think operation_type 0 is used.
https://github.com/bcgov/lear/blob/5260ebb3b8625e7e6d5bf05c555ea15f3b1e2596/python/common/sql-versioning/tests/test_versioning.py#L125

[the remark this answers is not in the document]
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/models/amalgamating_business.py:67]: OK. No need to change. I remember seeing some comments in your previous PR few weeks back related to this.

**Comment by eason-pan-bc** [comment]
Please provide some examples / screenshots of what kind of versioning data you are seeing for create/insertion, update and delete scenarios. Thank you 😃

**Review by eason-pan-bc — COMMENTED** [review]
in Local env, can you please put a screenshots showing all unit tests passed when flag is off. Since at the current stage, we'd like to make sure switching on and off both work.
Thank you.

**Comment by kzdev420** [comment]
@eason-pan-bc @AimeeGao 

Screenshot 2024-11-22 at 2 07 19 AM

**Review by leodube-aot — APPROVED** [review]

**Review by eason-pan-bc — APPROVED** [review]
LGTM, thank you for providing examples 😃

**Review by AimeeGao — APPROVED** [review]
LGTM(^_^)

**Pull request merged**